benchmark/torch/DT/evaluate_episodes.py

```diff
@@ -0,0 +1,100 @@
+# Third Party Code
+# The following code is copied or modified from:
+# https://github.com/kzl/decision-transformer/tree/master/gym/decision_transformer
+
+import numpy as np
+import torch
+from tqdm import tqdm
+
+
+def evaluate_episode_rtg(
+        env,
+        state_dim,
+        act_dim,
+        agent,
+        max_ep_len=1000,
+        scale=1000.,
+        state_mean=0.,
+        state_std=1.,
+        device='cuda',
+        target_return=None,
+):
+
+    state_mean = torch.from_numpy(state_mean).to(device=device)
+    state_std = torch.from_numpy(state_std).to(device=device)
+
+    state = env.reset()
+
+    # we keep all the histories on the device
+    # note that the latest action and reward will be "padding"
+    states = torch.from_numpy(state).reshape(1, state_dim).to(device=device, dtype=torch.float32)
+    actions = torch.zeros((0, act_dim), device=device, dtype=torch.float32)
+    rewards = torch.zeros(0, device=device, dtype=torch.float32)
+
+    ep_return = target_return
+    target_return = torch.tensor(ep_return, device=device, dtype=torch.float32).reshape(1, 1)
+    timesteps = torch.tensor(0, device=device, dtype=torch.long).reshape(1, 1)
+
+    sim_states = []
+
+    episode_return, episode_length = 0, 0
+    for t in range(max_ep_len):
+
+        # add padding
+        actions = torch.cat([actions, torch.zeros((1, act_dim), device=device)], dim=0)
+        rewards = torch.cat([rewards, torch.zeros(1, device=device)])
+
+        action = agent.predict(
+            (states.to(dtype=torch.float32) - state_mean) / state_std,
+            actions.to(dtype=torch.float32),
+            rewards.to(dtype=torch.float32),
+            target_return.to(dtype=torch.float32),
+            timesteps.to(dtype=torch.long),
+        )
+        actions[-1] = action
+        action = action.detach().cpu().numpy()
```

Comment:
maybe can put the tensor to numpy  in the `agent.py`.

Reply:
done